classes/village/VillageUpgrade.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+class VillageUpgrade {
+    public function __construct(
+        public int $id,
+        public string $key,
+        public int $village_id,
+        public string $status = VillageUpgradeConfig::UPGRADE_STATUS_LOCKED,
+        public ?int $research_progress = null,
+        public ?int $research_progress_required = null,
+        public ?int $research_progress_last_updated = null,
+        public bool $research_boosted = false,
+        private VillageUpgradeConfigData $config_data,
+    ) {}
+    public function getName(): string {
+        return $this->config_data->getName();
+    }
+    public function getTier(): int {
+        return $this->config_data->getTier();
+    }
+    public function getDescription(): string {
+        return $this->config_data->getDescription();
+    }
+    public function getResearchTime(): int {
+        return $this->config_data->getResearchTime();
```

Comment:
Where are you planning to account for research boosts?

Reply:
Research time is just the config value, so research boosts will be factored in two places:
- VillageUpgradeDto which passes a display string for the remaining time, we'll factor the village's research speed into that calculation and for the progress bar display on the front end.
- HourlyCron factors in research speed as a multiplier to the progress gained.